### PR TITLE
Docker and CI cleanup

### PR DESCRIPTION
**Pull request by ar13pit**
- Reduce the layers in the docker image
- CI scripts cleanup
- Execute CI scripts only in CI environment

### CODE REVIEW

**Comment by ar13pit** [comment]
I think because of the changes I have made to the Dockerfile, the `bash -c` in install-package.sh and build-package.sh may not be required and those docker exec commands can be cleaned up further

**Comment by MatthijsBurgh** [comment]
I am not experienced enough with Docker to say your changes are correct or to say that your changes are an improvement. So leaving this one completely up to @Timple

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 sudo: true
-dist: trusty
+dist: xenial
```
Comment (reinzor) [inline .travis.yml]: No official support for xenial in Travis ...

Comment (ar13pit) [reply .travis.yml]: https://docs.travis-ci.com/user/reference/overview/#virtualisation-environment-vs-operating-system

**Comment by reinzor** [comment]
Hmm that's new then 👍

**Comment by ar13pit** [comment]
@Timple @MatthijsBurgh Please review and merge this PR.

The changes/additions I have made with respect to the Docker file are to make the building of tue-env image locally too. I have also reduced the layers of the image and did some cleanup to the CI scripts.

I have thoroughly tested the docker image. So if either of you aren't going to review that part, then please just review the bash scripts and merge the PR.

**Comment by MatthijsBurgh** [comment]
Not experienced enough, so not reviewing.

**Comment by ar13pit** [comment]
@jlunenburg Yes it works perfectly fine. Have a look:
https://travis-ci.org/tue-robotics/grammar_parser/builds/500030986
https://travis-ci.org/tue-robotics/ed/builds/500032428

The build is same as that of master

**Comment by ar13pit** [comment]
After merge change the image tag in the Dockerfile in kaldi repository to master.

**Pull request merged**